lessons/ro/chapter_10.yaml

```
@@ -0,0 +1,145 @@
+
+- title: Ieșirea Standard (stdout)
+  content_markdown: >
+    Îți amintești de prima lecție? Poți observa ceva relevant legat de `I/O`?
+
+    Desigur, ceea ce face `println!` este o operațiune de output,
+    de fapt, direcționează (afișează) textul către `stdout`
+    și acesta va fi afișat pe ecran.
+
+    Dacă nu dorești să afișezi caracter de linie nouă `\n`, folosește `print!`.
+    code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use+std%3A%3Aio%3B%0A%0Afn+main%28%29+%7B%0A++++let+mut+input+%3D+String%3A%3Anew%28%29%3B%0A%0A++++%2F%2F+citirea+va+fi+oprita+la+apri%C8%9Bia+caracterului+%60%5Cn%60%0A++++if+let+Ok%28_%29+%3D+io%3A%3Astdin%28%29.read_line%28%26mut+input%29+%7B%0A++++++++println%21%28%22Textul+de+intrare+%3A+%7B%7D%22%2C+input%29%3B%0A++++%7D%0A%7D%0A
+
+- title: Eroarea Standard (stdout)
+  content_markdown: >
+    Pentru a separa raportarea erorilor de tipărirea obișnuită,
+    poți utiliza macro-urile `eprint!` și `eprintln!` care vor
+    afișa text pe canalul de eroare standard (`stderr`),
+    în loc de `stdout`.
+    Utilizați aceast macro cu un mesaj informativ.
+
+    În sisteme asemănătoare UNIX, cum ar fi macOS sau LINUX,
+    poți separa cele două tipuri de ieșire prin utilizarea redirecționărilor:
+    - `./main > output.txt`
+    - `./main >> output.txt`
+    - `./main 2> output.txt`
+    - `./main 2>> output.txt`
+    
+    > Comenzile cu `2` vor copia doar erorile generate de program, iar cele fără `2` vor elimina toate erorile.
+
+    > Comanda cu `>>` va adăuga text la sfârșitul fișierului, în timp ce operatorul `>` va suprascrie fișierul.
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use+std%3A%3Aio%3B%0A%0Afn+main%28%29+%7B%0A++++let+mut+input+%3D+String%3A%3Anew%28%29%3B%0A%0A++++%2F%2F+citirea+va+fi+oprita+la+apri%C8%9Bia+caracterului+%60%5Cn%60%0A++++if+let+Ok%28_%29+%3D+io%3A%3Astdin%28%29.read_line%28%26mut+input%29+%7B%0A++++++++println%21%28%22Textul+de+intrare+%3A+%7B%7D%22%2C+input%29%3B%0A++++%7D%0A%7D%0A
+
+- tile: Descriptori de Fișiere
+  content_markdown: >
+    Acum că știm care sunt operațiunile de bază `I/O`, intrăm mai în detaliu.
+
+    Ai mai văzut deja: `stdin` (intrare standard), `stdout` (ieșire standard) și `stderr` (eroare standard).
+    Pentru fiecare dintre ele i se asociază un număr întreg pozitiv, un identificator unic
+    pentru un canal `I/O` (de exemplu: fișier), cunoscut sub numele de `descriptor de fișier (fd)`.
+
+    Prin urmare:
+    - `stdin`: 0
+    - `stdout`: 1
+    - `stderr`: 2
+    - fișiere
+
+    [![file descriptors](https://cs-pub-ro.github.io/operating-systems/assets/images/file-descriptors-d19424f0a417ecd1c032f98a1969ad75.svg)
+
+    Lucrul cu fișiere joacă un rol important în operațiunile `I/O`.
+
+    Ați învățat deja să deschideți un fișier text și să citiți conținutul acestuia.
+
+    Dar ce zici de a scrie date în el, ca un canal `I/O`.
```

```suggestion
    Dar ce zici de a scrie date în el, folosindu-l pe post de canal `I/O`?
```
Aici exprimarea mi se pare oricum fortata, poate gasesti o expresie mai potrivita.